backend/routes.js

```diff
@@ -71,6 +71,16 @@ const route = new Proxy({}, {
   }
 })
 
+// helper function that returns 404 and prevents client from caching the 404 response
+// which can sometimes break things: https://github.com/okTurtles/group-income/issues/2608
+function notFoundNoCache (h) {
+  return h.response()
+    .code(404)
+    .header('Cache-Control', 'no-store, must-revalidate, proxy-revalidate')
+    .header('Pragma', 'no-cache')
+    .header('Expires', '0')
```

Comment:
`Expires` is supposed to have a timestamp, so `0` isn't a valid value. However, it's not really relevant if `cache-control` is present, except for clients that don't support HTTP/1.1. I don't think any such clients exist (today) to any significant, and even if they did, they can be ignored.